src/app/pushclient.js

```diff
@@ -1,86 +1,129 @@
-        qs: { documentId: document.documentId },
-        method: 'PUT',
-        headers: {
-          'content-type': 'application/json',
-          'Authorization': 'Bearer ' + this.pushtoken
-        },
-        body: JSON.stringify(document)
-      }, (error, response, body) => {
+      let params = this.getRequestParams(document);
+      request(params, (error, response/*, body*/) => {
         if (error) {
-          if (this.RETRY && retryCount < this.RETRY_ATTEMPTS) {
-            console.log(`Error || ${error} || trying to push ${document.documentId}...`);
-            console.log(`Retrying ${retryCount+1} time(s) to push ${document.documentId}...`);
-            return setTimeout(() => {this.pushDocumentPromise(document, ++retryCount)}, this.RETRY_DELAY)
-          } else {
-            reject(error);
-          }
+          // just trying to reduce code duplication... Just a proposal:
+          //
+
+          this.tryAgain(error, document, ++retryCount);
+
+          // if (this.RETRY && retryCount < this.RETRY_ATTEMPTS) {
+          //   console.log(`Error || ${error} || trying to push ${document.documentId}...`);
+          //   console.log(`Retrying ${retryCount+1} time(s) to push ${document.documentId}...`);
+          //   return setTimeout(() => {this.pushDocumentPromise(document, ++retryCount);}, this.RETRY_DELAY);
+          // }
+
+          reject(error);
+          return;
         }
-        if(response && response.statusCode != 202) {
-          if (this.RETRY && retryCount < this.RETRY_ATTEMPTS) {
-            console.log(`Error || ${response.statusCode} : ${response.body} || trying to push ${document.documentId}...`);
-            console.log(`Retrying ${retryCount+1} time(s) to push ${document.documentId}...`);
-            return setTimeout(() => {this.pushDocumentPromise(document, ++retryCount)}, this.RETRY_DELAY)
-          } else {
-            reject(`${response.statusCode} : ${response.body}`);
-          }
+        if(response && response.statusCode !== 202) {
+          this.tryAgain(`${response.statusCode} : ${response.body}`, document, ++retryCount);
+          // if (this.RETRY && retryCount < this.RETRY_ATTEMPTS) {
+          //   console.log(`Error || ${response.statusCode} : ${response.body} || trying to push ${document.documentId}...`);
+          //   console.log(`Retrying ${retryCount+1} time(s) to push ${document.documentId}...`);
+          //   return setTimeout(() => {this.pushDocumentPromise(document, ++retryCount);}, this.RETRY_DELAY);
+          // }
+
+          reject(`${response.statusCode} : ${response.body}`);
+          return;
         }
-        if(response && response.statusCode == 202) {
+        if(response && response.statusCode === 202) {
           resolve(`Pushed document : ${document.documentId}`);
+          // should you return here?
+          // Currently, both the resolve() and the reject() are called in this case.
```

Comment:
I don't think so? this is not what I observed, but I was not calling another function for the tryagain, so maybe now we need to?